src/components/new-safe/create/steps/ReviewStep/index.tsx

```diff
@@ -77,7 +77,7 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
   }
 
   const createSafe = async () => {
-    if (!wallet || !provider || !chain) return
+    if (!wallet || !provider || isWeb3ReadOnly(provider) || !chain) return
```

Comment:
I saw that we only use this condition here. How do we handle this case for normal tx executions i.e. how do we make sure that its not a read-only provider before pressing submit? Maybe we should try to apply the same logic to keep it consistent.

Reply:
> I saw that we only use this condition here

It's used in two other places: Safe creation retry and getting the MultiSend contract to submit a batch.

> How do we handle this case for normal tx executions i.e. how do we make sure that its not a read-only provider before pressing submit? Maybe we should try to apply the same logic to keep it consistent.

It's not necessary to include everywhere as you cannot submit when the wallet isn't connected. [Our `CheckWallet` component handles that](https://github.com/safe-global/safe-wallet-web/blob/consistent-provider/src/components/common/CheckWallet/index.tsx#L26-L27).

Reply:
Should we add the CheckWallet component to the submit buttons then instead of having this check?

Reply:
> Should we add the CheckWallet component to the submit buttons then instead of having this check?

A close equivalent is being used. The creation cannot happen if the wallet [is on the wrong chain](https://github.com/safe-global/safe-wallet-web/blob/trim-token-amount/src/components/new-safe/create/steps/ReviewStep/index.tsx#L200).

In this context, this would always be needed as it is a type guard for [`computeNewSafeAddress`](https://github.com/safe-global/safe-wallet-web/blob/dev/src/components/new-safe/create/logic/index.ts#L75).